contracts/Account.cairo

```diff
@@ -190,51 +192,99 @@ func execute{
         range_check_ptr, 
         ecdsa_ptr: SignatureBuiltin*
     }(
-        to: felt,
-        selector: felt,
-        calldata_len: felt,
-        calldata: felt*,
-        nonce: felt
-    ) -> (response : felt):
+        messages_len: felt,
```

Comment:
I think we should introduce a new `Call` struct containing only the `to` `selector` and `calldata` to avoid passing the `sender` and `nonce` which are global to all the calls, e.g:
```
struct Call:
   member to: felt
   member selector: felt
   member calldata_size: felt
   member calldata: felt*
end
``` 
and pass an array of `Call` to the `execute` method.

Reply:
That was my idea with the new `Message` structure that dropped `sender` but I didn't realize it can also drop the `nonce`, great call! (pun intended).

This was a draft only and it got stalled due to the reasons explained above but my idea was to eventually open a github discussion about it. Now that I have you here: what do you think about [having a single entrypoint](https://github.com/OpenZeppelin/cairo-contracts/discussions/27#discussioncomment-1548091)?

Pros:
* less code
* easier to reason about
* easier to implement

Cons:
* we'd be moving away from the interface agreed on https://github.com/OpenZeppelin/cairo-contracts/discussions/41
* I don't like the feel of having such a complex structure (array of structs) as the main standard interface, but can't really explain why


The alternative would be to keep the agreed interface and just add a `multi_execute` method or something like that, although it implies added complexity it'd be less lean.

Reply:
If we by chance decided to go for a new interface, we should push for it asap to make the relevant changes across tools (`Signer.py`, `starknet.js`, etc) while the ecosystem is still small. If that's the case, we would also need to push for Cairo to provide the required features.

Reply:
That or just add a separate method.

Reply:
Personally I like the idea of having a single entry point. I find it cleaner, but also I would find it weird to have 2 different methods to achieve the same thing (1 single call).  And maybe it will make it easier to deal with fees when that's added to StarkNet.

Changing the Account interface is fine since we are still very early. But we should check with the Starkware team when Cairo will support arrays of structs.

Reply:
Also, it would be interesting to understand the overhead of doing a single call when there is one entry point.

Reply:
Great. I let the starware team know this is blocking the standard. I'm under the impression that this interface will work and that if there's such an overhead, it can be worked out although input from the SW team would also be appreciated.